functests/performance/rt-kernel.go

```diff
@@ -59,4 +60,53 @@ var _ = Describe("[performance]RT Kernel", func() {
 
 	})
 
+	It("[test_id:28526][crit:high][vendor:cnf-qe@redhat.com][level:acceptance] Non worker-cnf node should not have RT kernel installed", func() {
+		var newWorkerNode *corev1.Node
+		newRole := "worker-test"
+		newLabel := fmt.Sprintf("%s/%s", testutils.LabelRole, newRole)
+
+		By("Skipping test if cluster does not have another available worker node")
+		workerNodes, err := nodes.GetByRole(testclient.Client, "worker")
```

Comment:
maybe try utilizing this syntax here:
` oc get node --selector='!node-role.kubernetes.io/worker-rt , node-role.kubernetes.io/worker'`

If `nodes.GetByRole` is not enough for this purpose you can add another util function under nodes

Reply:
we have the same code in updating_profile.go:
https://github.com/openshift-kni/performance-addon-operators/blob/a3ff53ae8ce51c5ed07b5143fdded78545512b9f/functests/performance/updating_profile.go#L128

I agree - probably it's better to create function in utils/nodes that will return the list of non-rt workers